### ISSUE TITLE
multiple bugs with goliath dna infusion

### ISSUE THREAD

**Issue by Serxule** [tgstation/tgstation#72350]
## Reproduction: 
No o2 amount is never enough except for being on lavaland and station atmosphere, any mix of an airtank will always say "(choking) not enough o2" unless you manually set the airtank pressure to 30kpa instead of 16kpa, and the 80 extra damage for lavaland fauna doesnt consistently work (or just stops working after some time), twice i was able to 1 hit a legion (which is the correct damage), but 2 other times it took 4+ hits to kill

**Comment by Floofies** [comment]
Looking into this issue, some of the lungs issues may be fixed by #71940

**Comment by Floofies** [comment]
Looking at the bonus damage issue, it is obvious what the bug is to me. This code eliminates the bonus permanently whenever you strike something with no density, or if it's something on the same turf as the player. I'm making another PR now to fix this and the other bugs!
```/obj/item/goliath_infuser_hammer/attack(mob/living/target, mob/living/carbon/human/user)
	if(!target.density || get_turf(target) == get_turf(user))
		faction_bonus_force = 0```
